src/main/java/com/jcraft/jsch/UserAuthKeyboardInteractive.java

```diff
@@ -193,4 +204,9 @@ public boolean start(Session session) throws Exception {
     }
     // return false;
   }
+
+  private boolean isMaxFailuresReached(int initial, int current) {
+    return Boolean.parseBoolean(JSch.getConfig("user_auth_keyboard_interactive_single_attempt"))
```

Comment:
This needs to use `session.getConfig()`, not `JSch.getConfig()` so that it can be set on a per Session basis.

I believe you also need to add the config item to `Session.applyConfig()` via a `Session.checkConfig()` call.
See https://github.com/mwiede/jsch/blob/d064f44b5ae4c9447908826df8f5f5eba7ee5a84/src/main/java/com/jcraft/jsch/Session.java#L3207.